homeassistant/components/braviatv/coordinator.py

```diff
@@ -0,0 +1,270 @@
+"""Update coordinator for Bravia TV integration."""
+from __future__ import annotations
+
+from collections.abc import Awaitable, Callable, Coroutine, Iterable
+from datetime import timedelta
+from functools import wraps
+import logging
+from typing import Any, Final, TypeVar
+
+from pybravia import BraviaTV, BraviaTVError
+from typing_extensions import Concatenate, ParamSpec
+
+from homeassistant.components.media_player.const import (
+    MEDIA_TYPE_APP,
+    MEDIA_TYPE_CHANNEL,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.debounce import Debouncer
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import CLIENTID_PREFIX, DOMAIN, NICKNAME
+
+_BraviaTVCoordinatorT = TypeVar("_BraviaTVCoordinatorT", bound="BraviaTVCoordinator")
+_P = ParamSpec("_P")
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL: Final = timedelta(seconds=10)
+
+
+def catch_braviatv_errors(
+    func: Callable[Concatenate[_BraviaTVCoordinatorT, _P], Awaitable[None]]
+) -> Callable[Concatenate[_BraviaTVCoordinatorT, _P], Coroutine[Any, Any, None]]:
+    """Catch BraviaTV errors."""
+
+    @wraps(func)
+    async def wrapper(
+        self: _BraviaTVCoordinatorT,
+        *args: _P.args,
+        **kwargs: _P.kwargs,
+    ) -> None:
+        """Catch BraviaTV errors and log message."""
+        try:
+            await func(self, *args, **kwargs)
+        except BraviaTVError as err:
+            _LOGGER.error("Command error: %s", err)
```

Comment:
Should we set the coordinator as not connected here and request a refresh?

Reply:
I wouldn't change "connected" here, as a conditionally unsupported command sent via the `remote.send_command` would result in a state change. 

In addition, after each command, a refresh request is called and in case of a fail, it will change the connection state:

https://github.com/home-assistant/core/blob/1c2aedc17d6f3a36f1630443aad112c86c4a25b9/homeassistant/components/braviatv/coordinator.py#L119-L126

https://github.com/home-assistant/core/blob/1c2aedc17d6f3a36f1630443aad112c86c4a25b9/homeassistant/components/braviatv/coordinator.py#L196-L200

But I think moving the call `await self.async_request_refresh()` to a decorator to reduce the number of code repeats a good idea.

Reply:
Done